priv/posts/marketing/20241121_getting-to-know-us-kiko-fernandez-reyes.md

```diff
@@ -0,0 +1,93 @@
+
+Personally, I think the use of documentation attributes in Erlang is extremely useful. I’ve worked with some parts of it, which is why I believe it’s so important. Before documentation attributes, we had edoc — which we still maintain — but it’s not something the Erlang/OTP team actively uses for our own documentation. We also had an XML format for generating Erlang documentation, but it was a bit tricky to work with. Ultimately, switching to documentation attributes and writing everything in Markdown was a great idea for producing better documentation. There are still some corner cases that we would like to fix in the future.
+
+Another highlight, of course, is the JIT. In many languages, there’s something called a tracing JIT. The JVM, for example, uses a tracing JIT, where it checks at runtime whether it’s worth optimizing certain patterns. However, Erlang doesn’t do that. Instead, we use something called a template-based JIT. I don’t think there are many programming languages out there doing the same.
+
+When it comes to Elixir, I’ve been following the Phoenix framework from the sidelines, and I think it’s excellent for web development. If you’re working in web development, I’d certainly recommend using Elixir. One notable recent update is the release of type inference, which I believe is due to their integration and experiments with set-theoretic types. Coming from a background in typed theory, I find this really important. I’m excited to see how this evolves over time.
+
+#### As a core member of the Erlang/OTP team, can you give us some insights into your current work and the challenges you face in this team? What does the Erlang/OTP team do?
+
+I work on the Erlang/OTP team at Ericsson. This team is responsible for developing and maintaining the Erlang programming language (including the compiler, virtual machine, etc.) as well as the set of libraries that come with it (OTP), such as the SSL library, gen_servers, and more. These libraries are designed to help developers create robust and fault-tolerant applications.
+
+As a core member of the Erlang/OTP, I worked a bit on the documentation attributes, but Lukas Larsson is the one that did all the heavy lifting, so all glory should go to him. From my side, there was a need to create a Markdown parser, because the Erlang emulator (shell) needs to render the documentation. Since Markdown is used for documentation attributes, the goal was to display the Markdown nicely in the Erlang shell. To achieve this, we parse the Markdown into an abstract syntax tree (AST) with a structure similar to HTML. This structure is what the internals of the shell understand when displaying documentation. The transformation from Markdown to HTML-AST is not exactly the same as what we used to have, but the parser still produces pretty accurate HTML-AST and we did not have to re-write the internals of the shell..

```

Comment:
```suggestion
As a core member of the Erlang/OTP, I worked a bit on the documentation attributes, but Lukas Larsson is the one that did all the heavy lifting, so all glory should go to him. From my side, there was a need to create a Markdown parser, because the Erlang emulator (shell) needs to render the documentation. Since Markdown is used for documentation attributes, the goal was to display the Markdown nicely in the Erlang shell. To achieve this, we parse the Markdown into an abstract syntax tree (AST) with a structure similar to HTML. This structure is what the internals of the shell understand when displaying documentation. The transformation from Markdown to HTML-AST is not exactly the same as what we used to have, but the parser still produces pretty accurate HTML-AST and we did not have to re-write the internals of the shell.

```